README.md

```diff
@@ -1,2 +1,115 @@
-# terraform-aws-eks-cluster
-This repository showcases the terraform we use to setup EKS cluster on AWS.
+# Introduction
+This repository showcases the terraform template that will help you to create EKS cluster on AWS. 
+
+---
+
+# AWS EKS Architecture
+![github_eks](https://user-images.githubusercontent.com/38158144/60167519-e29fa700-9820-11e9-9ecc-86be99973cd7.png)
+
+**Note** - Above architecture doesn't reflect all the components that are created by this template. However, it does give an idea about core infrastructure that will be created. 
+
+- Creates a new VPC with CIDR Block - 10.15.0.0/19 (i.e 8190 IPs in a VPC) in a region of your choice. Feel free to change it, values are `variables.tf`.
+- Creates 3 public & 3 private subnets with each size of 1024 IP addresses in each zones
+- Creates security groups required for cluster and worker nodes.
+- Creates recommened IAM service and EC2 roles required for EKS cluster.
+- Creates Internet & NAT Gateway required for public and private communications.
+- Routing Table and routes for public and private subnets.
+
+
+### Before you start
+Before you execute this template make sure following dependencies are met.
+
+- [Install terraform](https://releases.hashicorp.com/terraform/0.11.13/)
+- [Configure AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-linux-al2017.html) - make sure you configure AWS CLI with admin previliges 
+- [AWS iam authenticator](https://docs.aws.amazon.com/eks/latest/userguide/install-aws-iam-authenticator.html) - Amazon EKS uses IAM to provide authentication to your Kubernetes cluster through the AWS IAM Authenticator for Kubernetes.
+
+
+### Setup
+```
+$ git clone https://github.com/MediaIQ/terraform-aws-eks-cluster.git
+$ cd terraform-aws-eks-cluster
+```
+
+#### Initialize Terraform
+```
+$ terraform init
+```
+
+#### Terraform Plan
+The terraform plan command is used to create an execution plan. Always a good practice to run it before you apply it to see what all resources will be created.
+
+This will ask you to specify `cluster name` and worker node instance type. 
+
+```
+$ terraform plan
+
+var.cluster-name
+  Enter eks cluster name - example like eks-demo, eks-dev etc
+
+  Enter a value: eks-demo
+
+var.region
+  Enter region you want to create EKS cluster in
+
+  Enter a value: us-east-1
+
+var.ssh_key_pair
+  Enter SSH keypair name that already exist in the account
+
+  Enter a value: eks-keypair
+
+var.worker-node-instance_type
+  enter worker node instance type
+
+  Enter a value: t2.medium
+
+```
+
+#### Apply changes
+```
+$ terraform apply
+```
+
+#### Configure kubectl
+```
+$ aws eks --region eu-central-1 update-kubeconfig --name <CLUSTER-NAME>
```

Comment:
Can you please change this to:

`aws eks --region <AWS-REGION> update-kubeconfig --name <CLUSTER-NAME>`

Reply:
yo

Reply:
done